src/set_theory/cardinal/basic.lean

```diff
@@ -298,6 +298,9 @@ theorem le_one_iff_subsingleton {α : Type u} : #α ≤ 1 ↔ subsingleton α :=
 ⟨λ ⟨f⟩, ⟨λ a b, f.injective (subsingleton.elim _ _)⟩,
  λ ⟨h⟩, ⟨⟨λ a, punit.star, λ a b _, h _ _⟩⟩⟩
 
+lemma _root_.set.subsingleton.cardinal_mk_le_one {s : set α} (hs : s.subsingleton) : #s ≤ 1 :=
```

Comment:
The `iff` version of this would be nice

Reply:
even given that it's just `le_one_iff_subsingleton.trans set.subsingleton_coe`?

Reply:
Oh I guess the middle one is a class, so it does make sense.